src/tools/opt-dist/src/training.rs

```diff
@@ -66,20 +66,32 @@ fn init_compiler_benchmarks(
     .workdir(&env.rustc_perf_dir())
 }
 
+/// Describes which `llvm-profdata` binary should be used for merging PGO profiles.
+enum LlvmProfdata {
+    /// Use llvm-profdata from the host toolchain (i.e. from LLVM provided externally).
+    Host,
+    /// Use llvm-profdata from the target toolchain (i.e. from LLVM built from `src/llvm-project`).
+    Target,
+}
+
 fn merge_llvm_profiles(
     env: &dyn Environment,
     merged_path: &Utf8Path,
     profile_dir: &Utf8Path,
+    profdata: LlvmProfdata,
 ) -> anyhow::Result<()> {
-    cmd(&[
-        env.downloaded_llvm_dir().join("bin/llvm-profdata").as_str(),
-        "merge",
-        "-o",
-        merged_path.as_str(),
-        profile_dir.as_str(),
-    ])
-    .run()
-    .context("Cannot merge LLVM profiles")?;
+    let llvm_profdata = match profdata {
+        LlvmProfdata::Host => env.downloaded_llvm_dir().join("bin/llvm-profdata"),
```

Comment:
Side note: I found this name somewhat confusing, because I initially thought it refers to download-ci-llvm, not host llvm.

Reply:
Yeah, the name could be better, but I'm not sure what it should be. For Linux, it is the LLVM coming from the Dockerfile. However, for Windows, it is an actually downloaded LLVM that comes from S3 (it has nothing to do with `download-ci-llvm` though).

Reply:
In both cases it would still be the host LLVM, so maybe `host_llvm_dir` or similar would convey the intent?

Reply:
Ok, I'll send a PR with a rename once this one gets merged.